contrib/src/main/java/com/datatorrent/contrib/parser/RegexParser.java

```diff
@@ -0,0 +1,234 @@
+
+/**
+ * Operator parsers the tuple based on regex pattern and populates POJO matching the user defined schema <br>
+ * This operator expects the upstream operator to send every line in the file as a byte array.
+ * splitRegexPattern contains the regex pattern of lines in the file <br>
+ * Schema is specified in a json format as per {@link DelimitedSchema} that
+ * contains field information and constraints for each field.<br>
+ * Schema field names should match with the POJO variable names<br>
+ * Assumption is that each field in the delimited data should map to a simple
+ * java type.<br>
+ * <br>
+ * <b>Properties</b> <br>
+ * <b>splitRegexPattern</b>:Regex pattern as a string<br>
+ *
+ * @displayName RegexParser
+ * @category Parsers
+ * @tags pojo parser regex logs server
+ * @since 3.6.0
+ */
+@InterfaceStability.Evolving
+public class RegexParser extends Parser<byte[], KeyValPair<String, String>>
+{
+  /**
+   * Contents of the schema.Schema is specified in a json format as per
+   * {@link DelimitedSchema}
+   */
+  @NotNull
+  private String schema;
+  /**
+   * Schema is read into this object to access fields
+   */
+  private transient DelimitedSchema delimitedParserSchema;
+  /**
+   * Regex Pattern defined for the tuple
+   */
+  @NotNull
+  private String splitRegexPattern;
+  /**
+   * Pattern to store the compiled regex
+   */
+  private transient Pattern pattern;
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    delimitedParserSchema = new DelimitedSchema(schema);
+    pattern = Pattern.compile(splitRegexPattern);
+  }
+
+  @Override
+  public void processTuple(byte[] tuple)
+  {
+    if (tuple == null) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<String, String>(null, "Blank/null tuple"));
+      }
+      errorTupleCount++;
+      return;
+    }
+    String incomingString = new String(tuple);
+    if (StringUtils.isBlank(incomingString)) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<String, String>(incomingString, "Blank tuple"));
+      }
+      errorTupleCount++;
+      return;
+    }
+    try {
+      if (out.isConnected() && clazz != null) {
+        Matcher matcher = pattern.matcher(incomingString);
+        boolean patternMatched = false;
+        Constructor<?> ctor = clazz.getConstructor();
+        Object object = ctor.newInstance();
+
+        while (matcher.find()) {
+          for (int i = 1; i <= matcher.groupCount(); i++) {
+            if (delimitedParserSchema.getFields().get(i - 1).getType() == DelimitedSchema.FieldType.DATE) {
+              DateTimeConverter dtConverter = new DateConverter();
+              dtConverter.setPattern((String)delimitedParserSchema.getFields().get(i - 1).getConstraints().get(DelimitedSchema.DATE_FORMAT));
+              ConvertUtils.register(dtConverter, Date.class);
+            }
+            BeanUtils.setProperty(object, delimitedParserSchema.getFields().get(i - 1).getName(), matcher.group(i));
```

Comment:
PojoUtils doesn't support Date fields. Using PojoUtils for the rest and Reflection for Date can be tried, if that is feasible solution I will track it as a different issue. At present, a single instance of an operator is able to process 32K records per second with 2 strings, 1 long and 1 Date field. The test case has the String and the regex pattern.